Extensions/xkit_patches.js

```diff
@@ -599,6 +681,80 @@ XKit.extensions.xkit_patches = new Object({
 
 			XKit.interface.post_window.reblogging_from =
 				() => $(".post-form--header .reblog_source .reblog_name").text();
+
+			/**
+			 * @param {JQuery} obj - Post element
+			 * @return {Promise<Object>} Resolves to an interface Post Object or rejects
+			 */
+			XKit.interface.async_post = function(obj) {
+				if ($(obj).attr('data-id') && XKit.page.react) {
+					return XKit.interface.react.post(obj);
+				} else {
+					const post_object = XKit.interface.post(obj);
+					if (post_object.error) {
+						return Promise.reject(post_object);
+					} else {
+						return Promise.resolve(post_object);
+					}
+				}
+			};
+
+			XKit.interface.react = {
+				api_fetch: async function(path, options = {}) {
+					// eslint-disable-next-line no-shadow
+					return XKit.tools.async_add_function(({path, options, xkit_version}) => {
+						/* globals tumblr */
```

Comment:
i'd prefer if this were in the actual eslint file, especially since we're injecting it

Reply:
@AprilSylph it can't be in the actual eslint file, it's only global in this one block

Reply:
fair

Reply:
(also this is our standard practice for injected functions)